node/tessel-export.js

```diff
@@ -952,6 +957,107 @@ Tessel.LED.prototype.read = function(callback) {
   });
 };
 
+Tessel.Network = {
+  Wifi: function() {
+    var state = {
+      settings: {},
+      connected: false
+    };
+
+    Object.defineProperties(this, {
+      isConnected: {
+        get: function() {
+          return !!state.connected;
+        }
+      },
+      connected: {
+        set: function(value) {
+          state.connected = value;
+        }
+      },
+      settings: {
+        get: function() {
+          return state.settings;
+        },
+        set: function(settings) {
+          util.inherits(state.settings, settings);
+        }
+      }
+    });
+  }
+};
+
+Tessel.Network.Wifi.prototype.connect = function(settings, callback) {
+  if (typeof settings !== 'object' || settings.ssid.length === 0) {
+    throw new Error('Wifi settings must be an object with at least a "ssid" property.');
+  }
+
+  if (typeof callback !== 'function') {
+    callback = function() {};
+  }
+
+  if (settings.password && !settings.security) {
+    settings.security = 'psk2';
+  }
+
+  if (!settings.password && (!settings.security || settings.security === 'none')) {
+    settings.security = 'none';
+  }
+
+  var self = this;
+  connectToNetwork(settings)
+    .then(commitWireless)
+    .then(restartWifi)
+    .then(function() {
+      self.settings = settings;
+      callback(null, self.settings);
+    })
+    .catch(function(err) {
+      callback(err);
+    });
+};
+
+function connectToNetwork(settings) {
+  var commands = [
+    'uci batch <<EOF',
+    'set wireless.@wifi-iface[0].ssid=' + settings.ssid,
+    'set wireless.@wifi-iface[0].key=' + settings.password,
+    'set wireless.@wifi-iface[0].encryption=' + settings.security,
+    'EOF'
+  ];
```

Comment:
I wasn't sure what new features I was able to use. This is certainly much cleaner. Thanks. :+1: